src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Triangle sides must be greater than 0');
+    }
+
+    if (this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.b + this.a
+    ) {
+      throw new Error('Invalid triangle sides');
+    }
 
-export class Circle {
+    this.color = color;
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = 'circle';
+
+  constructor(
+    public color: string,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Circle radius must be greater than 0');
+    }
+
+    this.color = color;
```

Comment:
It's not necessary to assign 'color' and 'radius' to 'this.color' and 'this.radius' respectively, as they are already assigned using 'public' and 'private' keywords in the constructor.